apps/examples/wifi_manager_sample/wm_test.c

```diff
@@ -678,215 +676,297 @@ void wm_auto_test(void *arg)
+		printf("Connecting to AP (long SSID)\n");
+		res = wifi_manager_connect_ap(&ap_fake_config);
 		if (res != WIFI_MANAGER_SUCCESS) {
 			printf("AP connect failed due to bad configuration: long SSID\n");
 			res = WIFI_MANAGER_FAIL;
-			fake_config.ssid_length = strlen(fake_ssid);
-			strncpy(fake_config.ssid, fake_ssid, fake_config.ssid_length);
-			fake_config.ssid[fake_config.ssid_length + 1] = '\0';
+		} else {
+			WM_TEST_WAIT;
 		}
-		printf("Connecting to fake AP (Auth/Crypto type 2/3)\n");
+
 		/* Connect to fake AP */
-		res = wifi_manager_connect_ap(&fake_config);
+		printf("Connecting to AP (bad SSID)\n");
+		strncpy(ap_config.ssid, info->bad_ssid, WIFIMGR_SSID_LEN-1);
+		ap_config.ssid[WIFIMGR_SSID_LEN-1] = '\0';
+		ap_config.ssid_length = strlen(info->bad_ssid);
+		print_wifi_ap_profile(&ap_config, "Connecting AP Info");
+		res = wifi_manager_connect_ap(&ap_config);
 		if (res != WIFI_MANAGER_SUCCESS) {
-			printf("AP connect failed due to bad configuration: auth/crypto type 2/3\n");
+			printf("AP connect failed due to bad configuration: bad SSID\n");
 			res = WIFI_MANAGER_FAIL;
-			fake_config.ap_auth_type = 1;
-			fake_config.ap_crypto_type = info->crypto_type;
+		} else {
+			WM_TEST_WAIT;
+		}
+
+		/* Connect to fake AP */
+		printf("Connecting to AP (wrong passphrase)\n");
+		strncpy(ap_config.ssid, info->ssid, WIFIMGR_SSID_LEN-1);
+		ap_config.ssid[WIFIMGR_SSID_LEN-1] = '\0';
+		ap_config.ssid_length = strlen(info->ssid);
+		strncpy(ap_config.passphrase, info->bad_password, WIFIMGR_PASSPHRASE_LEN-1);
+		ap_config.passphrase[WIFIMGR_PASSPHRASE_LEN-1] = '\0';
+		ap_config.passphrase_length = strlen(info->bad_password);
+		print_wifi_ap_profile(&ap_config, "Connecting AP Info");
+		res = wifi_manager_connect_ap(&ap_config);
+		if (res != WIFI_MANAGER_SUCCESS) {
+			printf("AP connect failed due to bad configuration: bad passphrase\n");
+			res = WIFI_MANAGER_FAIL;
+		} else {
+			WM_TEST_WAIT;
+		}
+
+		/* Connect to fake AP */
+		printf("Connecting to fake AP (Auth/Crypto type 1/3)\n");
+		ap_config.ap_auth_type = WIFI_MANAGER_AUTH_WEP_SHARED;
+		strncpy(ap_config.passphrase, info->password, WIFIMGR_PASSPHRASE_LEN-1);
+		ap_config.passphrase_length = strlen(info->password);
+		ap_config.passphrase[WIFIMGR_PASSPHRASE_LEN-1] = '\0';
+		print_wifi_ap_profile(&ap_config, "Connecting AP Info");
+		res = wifi_manager_connect_ap(&ap_config);
+		if (res != WIFI_MANAGER_SUCCESS) {
+			printf("AP connect failed due to bad configuration: auth/crypto type 1/3\n");
+			res = WIFI_MANAGER_FAIL;
+		} else {
+			WM_TEST_WAIT;
 		}
-		WM_TEST_WAIT;
 
-		printf("Connecting to AP\n");
 		/* Connect to AP */
+		ap_config.ap_auth_type = info->auth_type;
+		printf("Connecting to AP\n");
+		print_wifi_ap_profile(&ap_config, "Connecting AP Info");
 		res = wifi_manager_connect_ap(&ap_config);
 		if (res != WIFI_MANAGER_SUCCESS) {
 			printf("AP connect failed in round %d\n", cnt);
-			return ;
+			res = WIFI_MANAGER_FAIL;
```

Comment:
@pillip8282. For now, this is unnecessary assigment. Considering all your comments, it would be necessary to add a new command to measure code coverage instead of auto test. The update will be completed soon.